lib/src/presentation/bloc/similar_movies_bloc.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:async';
+
+import '../../core/bloc/movies_bloc_interface.dart';
+import '../../core/util/data_state.dart';
+import '../../core/util/enums.dart';
+import '../../core/util/strings.dart';
+import '../../domain/entity/event/implementation/movie_event.dart';
+import '../../domain/entity/movie.dart';
+import '../../domain/usecase/implementation/get_similar_movies_usecase.dart';
+
+class SimilarMoviesBloc implements IMoviesBloc<int> {
```

Comment:
All the blocs that you defined for SimilarMovies, SearchMovies, etc are all mostly the same, review all the blocs to see if you can wrapped in one